core/src/main/java/com/datastax/oss/driver/api/core/config/DriverExecutionProfile.java

```diff
@@ -37,7 +37,7 @@
  *
  * @see DriverConfig
  */
-public interface DriverExecutionProfile {
+public interface DriverExecutionProfile extends OngoingConfigOptions<DriverExecutionProfile> {
```

Comment:
What I like is that the list of all possible setters lives in a single interface. If we add a method on execution profiles, we won't have to remember to add it on the programmatic builder every time.

Reply:
Which means that we are back to the same situation as in `BoundStatementBuilder` vs `BoundStatement`: you will have to annotate all the methods in this interface with `@CheckReturnValue` to warn people but it won't be necessary sometimes to re-assign the returned value. OK, I guess I can live with that.

Reply:
Actually the methods were not annotated with `@CheckReturnValue` in `DriverExecutionProfile`. If I add the annotations now it will cause a Revapi violation.
Given that `withXxx` method are less ambiguous, and I don't expect derived profiles to be widely used, I prefer not adding another set of exclusions now.